controllers/authControll.js

```diff
@@ -33,12 +43,53 @@ const register = async (req, res) => {
   });
 };
 
+const verifyEmail = async (req, res) => {
+  const { verificationToken } = req.params;
+  const user = await User.findOne({ verificationToken });
+  if (!user) {
+    throw HttpError(404, "User not found");
+  }
+
+  await User.findByIdAndUpdate(user._id, {
+    verify: true,
+    verificationToken: "",
+  });
+
+  res.status(200).json({ message: "Verification successful" });
+};
+
+const resendVerifyEmail = async (req, res) => {
+  const { email } = req.body;
+  const user = await User.findOne({ email });
+  if (!user) {
+    throw HttpError(404, "User not found");
```

Comment:
Zamiast throw w tych najwyższych ścieżkach trzeba zwrócić błąd użytkownikowi. Bo zastanawiam się co przechwyci ten throw? Jeśli robisz tutaj throw to znaczy że gdzieś wyżej, czyli w miejscu gdzie jest użyta funkcja resendVerifyEmail jest blok catch. Jeśli tego bloku nie ma, to aplikacja przestanie działać. 

I to się tyczy wielu miejsc.